src/doc/specifying-dependencies.md

```diff
@@ -0,0 +1,304 @@
+% Specifying Dependencies
+
+Your crates can depend on other libraries from crates.io, git repositories, or
+subdirectories on your local file system. You can also temporarily override the
+location of a dependency-- for example, to be able to test out a bug fix in the
+dependency that you are working on locally. You can have different
+dependencies for different platforms, and dependencies that are only used during
+development. Let's take a look at how to do each of these.
+
+# Specifying Dependencies from crates.io
+
+Cargo is configured to look for dependencies on `crates.io` by default. Only
+the name and a version string are required in this case. In [the cargo
+guide](guide.html), we specified a dependency on the `time` crate:
+
+```toml
+[dependencies]
+time = "0.1.12"
+```
+
+The string `"0.1.12"` is a [semver][semver] version requirement. Since this
+string does not have any operators in it, it is interpreted the same way as
+if we had specified `"^0.1.12"`, which is called a caret requirement.
+
+[semver]: https://github.com/steveklabnik/semver#requirements
+
+## Caret requirements
+
+**Caret requirements** allow SemVer compatible updates to a specified version.
+An update is allowed if the new version number does not modify the left-most
+non-zero digit in the [major, minor, patch] tuple. In this case, if we ran
```

Comment:
slightly weird to call this a tuple with `[` instead of `(`, since Rust tuples use `(`...